File: backend/app/schemas/paper.py
from datetime import datetime

from pydantic import BaseModel, Field


class PaperBase(BaseModel):
    title: str = Field(min_length=2)
    authors: str | None = None
    journal: str | None = None
    year: str | None = None
    pmid: str | None = None
    doi: str | None = None
    source_url: str
    abstract: str | None = None


class PaperSave(PaperBase):
    pass


class PaperRead(PaperBase):
    id: int
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/paper.ts
export type Paper = {
  id?: number;
  title: string;
  authors?: string | null;
  journal?: string | null;
  year?: string | null;
  pmid?: string | null;
  doi?: string | null;
  source?: string;
  source_url: string;
  abstract?: string | null;
  score?: number;
  created_at?: string;
};

